logstash-core-event-java/src/main/java/com/logstash/RubyJavaObject.java

```diff
@@ -0,0 +1,124 @@
+package com.logstash;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.jruby.Ruby;
+import org.jruby.RubyNil;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.io.ObjectInputStream;
+
+public final class RubyJavaObject implements Serializable {
+    private static final long serialVersionUID = 4358368393114318978L;
+
+    private transient IRubyObject rubyValue;
+    private transient boolean isRubyNil;
```

Comment:
is there a particular reason why we keep a specific boolean for tagging nil values? it seems that `hasRubyValue() && rubyValue.isNil()` would work and avoid managing this boolean state?